iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift

```diff
@@ -111,15 +112,28 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
     }
     
     func set(state: ConversationUIState) {
-        self.collectionLayout.uiState = state
-        self.collectionLayout.prepareForTransition(to: self.collectionLayout)
-        
+        self.configureCollectionLayout(for: state)
+
         Task {
-            await UIView.awaitAnimation(with: .standard, animations: {
-                self.scrollToLastItemOnLayout = true
-                self.setNeedsLayout()
-                self.collectionLayout.finalizeLayoutTransition()
-            })
+            await self.dataSource.reconfigureAllItems()
+            if state == .write {
+                let maxOffset = self.collectionLayout.maxZPosition
+                self.collectionView.setContentOffset(CGPoint(x: 0, y: maxOffset), animated: true)
+            }
+        }
```

Comment:
Updating the layout parameters then reconfiguring the items makes the animations look smooth. Attempting to animation a with layoutNow/setNeedsLayout is less successful.